source/fundamentals/crud/write-operations/upsert.txt

```diff
@@ -4,8 +4,99 @@ Insert or Update in a Single Operation
 
 .. default-domain:: mongodb
 
-If your application stores and modifies data in MongoDB, you probably use
-insert and update operations. In certain workflows, you may need to choose
-between an insert and update depending on whether the document exists.
-In these cases, you can streamline your application logic by using the
-``upsert`` option available in the following methods:
+Overview
+--------
+
+Insert or update documents by passing a query filter to the 
+:java-docs:`updateOne() </apidocs/mongodb-driver-sync/com/mongodb/client/MongoCollection.html#updateOne(org.bson.conversions.Bson,org.bson.conversions.Bson,com.mongodb.client.model.UpdateOptions)>`, 
+:java-docs:`updateMany() </apidocs/mongodb-driver-sync/com/mongodb/client/MongoCollection.html#updateMany(org.bson.conversions.Bson,org.bson.conversions.Bson,com.mongodb.client.model.UpdateOptions)>`
+or :java-docs:`replaceOne() </apidocs/mongodb-driver-sync/com/mongodb/client/MongoCollection.html#replaceOne(org.bson.conversions.Bson,TDocument,com.mongodb.client.model.ReplaceOptions)>`
+methods. These methods update documents that match your query filter,
+but inserts a document if the there are no matches when you set the
+upsert to true. When multiple documents match the query filter supplied to
+``updateOne()`` or ``replaceOne()``, the first document that matches
+gets updated.
```

Comment:
🤔 

This part still feels off to me. If someone didn't know about upsert behavior, the introduction sentence might leave them feeling like they missed something earlier.

Maybe a gentler introduction that states documents can be inserted using update or replace using upsert behavior, and then how to specify upsert behavior, and then why it might be useful?